### DISCUSSION TITLE
Alternative Android version

### DISCUSSION THREAD

**Discussion by moukrea** [redphx/better-xcloud#363] [General]
Hi!

I've been spending some of my free time lately on making an Android app that packs up Better xCloud using WebView, only to find out it already exists! Anyhow, I still released my take on it and plan on improve upon it, you can retrieve it there: [Game Pass Plus](https://github.com/Moukrea/gamepassplus)

It'll soon be ported over Desktops (using a combination of Go + Wails or Electron depending on how it goes) and as a PWA.

Better xCloud is heavily credited in it, I do not intend to get all the credit 😉 

EDIT: Whoops, the release build contains an issue preventing Better xCloud to start (even though properly installed), which wasn't the case in the pre-release build (not available on GitHub), I'll fix that as soon as I get home! An issue [has been created](https://github.com/Moukrea/gamepassplus/issues/1)

**Comment by redphx** [comment]
Congrats on the release! Best of luck to your project.

**Comment by KingNothing81** [comment]
That looks interesting. I’m curious to give it a try. 

**Edit**: I tried it now. Had some problems logging in. Always got an “Ooops” error page. I then somehow managed to get to the login screen by using one of the Microsoft links at the bottom. But what I really like is that it’s really full screen without a tool or address bar at the top or bottom. Looks so much better on a TV. I hope the BXC app will let us have this too one day.

**Reply by moukrea** [reply]
The issue has been identified, it appeared in the latest patch release, that fixed Better xCloud injection and Auto-update. Incidentally, the (now) consistent script injection causes login issues and/or access to home screen issues when the session has been stale for too long

A fix is in the work, along with a fix to bypass the "Start playback" button when running a game that requires a mouse on TV to be bypassed

As for other issues, app dialog boxes have titles placement issues for devices with notches (when held vertically) and wrong available space usage when no buttons are visible (ie. the Better xCloud download screen)

Whilst not a priority, this will be addressed later on.

Then we'll move onto user experience improvements before moving onto other platforms.

Thanks for you feedback!

**Reply by KingNothing81** [reply]
Thanks for your answer. I'm a Android TV only user, so improving the experience on that platform has always been my personal, egositic priority. ;) Because at the moment you still need "mouse support" to do some things. So maybe you both can even join forces to make things more remote/controller friendly? 

> A fix is in the work, along with a fix to bypass the "Start playback" button when running a game that requires a mouse on TV to be bypassed

Regarding this, I noticed that it's also possible to get past that green "press to play" button by pressing the START (right side) button on the controller. So knowing that, you can at least get past this without a mouse. ;)

But I'm curious how your app will evolve in the future and keep an eye on it. :)

**Reply by moukrea** [reply]
We both use fairly different approach, but I think we'll get to common grounds later down the line when it'll come to user experience improvements ;)
Neat trick, didn't catch this one!
